core/rfb.js

```diff
@@ -604,7 +614,14 @@ export default class RFB extends EventTargetMixin {
 
     _updateScale() {
         if (!this._scaleViewport) {
-            this._display.scale = 1.0;
+            if (this._nativeDpi) {
+                const dpr = window.devicePixelRatio || 1;
+                const scaledWidth = Math.floor(window.innerWidth * dpr);
+                // Compute the exact scale factor to match the floored width/height
+                this._display.scale = window.innerWidth / scaledWidth;
```

Comment:
I don't understand the purpose of this calculation. Why not just `1 / window.devicePixelRatio`?